File: assignment_2/Model/Expressions/VarExpression.java
package Model.Expressions;

import ADTs.IDictionary;
import Exceptions.UndefinedVariableException;

public class VarExpression extends IExpression {
    private String variable;

    public VarExpression(String var) {
        this.variable = var;
    }

    public int eval(IDictionary<String, Integer> table) throws UndefinedVariableException {
        if(table.get(this.variable) != null)
            return table.get(this.variable);
        throw new UndefinedVariableException();
    }

    public String toString() {
        return this.variable;
    }
}

File: Toy_Language_Interpreter_C/Expressions/VarExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class VarExpression : IExpression
    {
        private String variable;
        
        public VarExpression(String var)
        {
            this.variable = var;
        }

        public override int Eval(IDictionary<string, int> table)
        {
            try
            {
                return table.Get(this.variable);
            }
            catch(UndefinedOperationException)
            {
                throw new UndefinedVariableException(this.variable);
            }
        }

        public override string ToString()
        {
            return this.variable;
        }
    }
}
